src/ut_control_plane.c

```diff
@@ -0,0 +1,394 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2023 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Standard Libraries */
+#include <pthread.h>
+#include <stdio.h>
+
+/* Application Includes */
+#include <ut_control_plane.h>
+#include <ut_kvp.h>
+
+/* External libraries */
+#include <libwebsockets.h>
+
+#define UT_CONTROL_PLANE_ERROR(f_, ...) printf((f_), ##__VA_ARGS__)
+#define UT_CONTROL_PLANE_DEBUG(f_, ...) (void)0
+
+typedef struct
+{
+  char key[UT_CONTROL_PLANE_MAX_KEY_SIZE];
+  ut_control_callback_t pCallback;
+}CallbackEntry_t;
+
+typedef struct
+{
+    uint32_t magic;
+    struct lws_context_creation_info info;
+    struct lws_context *context;
+    pthread_t thread;
+    pthread_t thread2;
+} ut_cp_instance_internal_t;
+
+typedef enum
+{
+    SERVER_START = 0,
+    EXIT_REQUESTED,
+    DATA_RECIEVED
+}eMessage_t;
+typedef struct
+{
+ eMessage_t status;
+ char *message;
+ uint32_t size;
+}cp_message_t;
+
+#define MAX_CALLBACKS 10
+#define MAX_MESSAGE_LEN 256
+#define MAX_MESSAGES 100
+
+typedef struct{
+    char key[100];
+    void (*callback)(void *);
+}ut_cp_callback_t;
+
+ut_cp_callback_t callback_list[MAX_CALLBACKS];
```

Comment:
duplicate list -> `CallbackEntry_t` is the entry for a list..

you appear to also have defined `ut_cp_callback_t` which is the same thing only not as good.